core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2267,4 +2267,12 @@ package object config {
       .version("3.3.0")
       .intConf
       .createWithDefault(5)
+
+  private[spark] val MEMORY_ONHEAP_PAGESIZE_OPTIMIZE_ENABLED =
```

Comment:
Do we need a flag for this? when would I not want it?

Reply:
yeah, it's indeed no need. removed it